xclim/indices/stats.py

```diff
@@ -710,7 +739,35 @@ def standardized_index_fit_params(
     Supported combinations of `dist` and `method` are:
     * Gamma ("gamma") : "ML", "APP", "PWM"
     * Log-logistic ("fisk") : "ML", "APP"
+    * "APP" method only supports two-parameter distributions. Parameter `loc` will be set to 0 (setting `floc=0` in `fitkwargs`).
+
+    When using the zero inflated option, : A probability density function :math:`\texttt{pdf}_0(X)` is fitted for :math:`X \neq 0`
+    and a supplementary parameter :math:`\pi` takes into account the probability of :math:`X = 0`. The full probability density
+    function is a piecewise function
+
+    .. math::
+      \texttt{pdf}(X) = \pi  \texttt{ if }  X=0  \texttt{ else } (1-\pi) \texttt{pdf}_0(X)
     """
+    fitkwargs = fitkwargs or {}
+    if method == "APP":
+        if "floc" in fitkwargs.keys():
+            if fitkwargs["floc"] != 0:
+                raise ValueError(
+                    "The APP method is only supported for two-parameter distributions with `gamma` or `fisk`. `loc` parameter must be set 0."
+                    "Pass `floc=0` in `fitkwargs`."
+                )
```

Comment:
I don't understand. Doesn't the implementation above actually  implement this ?

Reply:
Yes, we could in principle allow `APP` with `loc != 0`. This is just a concern about the quality of the fit: 
* This is taking into account how David was more or less on board with advertising `APP` method
* The way I implemented how to find the `loc` parameter is generic, independent of the distribution really, and by no means perfect. I tried finding a better formula with the method of moments and analytic approximations. It works very well with two parameters `shape, scale`, but not so much once the third parameter `loc` is added in. 

From what I have seen, I would not recommend using the method `APP` in this case

Reply:
But then how do you use APP with SPEI in xclim >= 0.50, when `offset` is removed ?

Reply:
As this the offset technique is what `climate_indices` does, I would let this option open, even if it's only through the `standardized_index_fit_params` method (but then, this would need to be documented!).

Reply:
Ok, I left the offset only in `standardized_index_fit_params`, and there are still some temporary checks in SPEI. 

What you had in mind for documentation? A notebook example to show how to reproduce `climate_indices`?

Reply:
I'm not sure... I would say simply an example of how to passe an offset, I thought it would be through `floc` in `fitkwargs`.

Reply:
I'm looking into this. Perhaps allowing `APP` only when `floc` is given (`loc` is fixed, so effectively 2-parameter distribution). I'll see if using `floc` as a way to have the offset gives the expected results

Reply:
Seems to be working fine, I'll go with this